examples/ogbn_train.py

```diff
@@ -0,0 +1,384 @@
+
+        return x_all
+
+
+if args.use_gat:
+    print("GAT model")
+    model = GAT(dataset.num_features, num_hidden_channels, dataset.num_classes,
+                num_layers=num_layers, dropout=args.dropout,
+                heads=args.num_heads)
+else:
+    print("GraphSage model")
+    model = SAGE(dataset.num_features, num_hidden_channels,
+                 dataset.num_classes, num_layers=num_layers,
+                 dropout=args.dropout)
+
+model = GAT(dataset.num_features, num_hidden_channels, dataset.num_classes,
+            num_layers=num_layers, dropout=args.dropout,
+            heads=args.num_heads).to(device)
+
+model = model.to(device)
+
+
+def train(epoch):
+    model.train()
+
+    pbar = tqdm(total=split_idx['train'].size(0))
+    pbar.set_description(f'Epoch {epoch:02d}')
+
+    total_loss = total_correct = 0
+    for batch in train_loader:
+        optimizer.zero_grad()
+        out = model(batch.x, batch.edge_index.to(device))[:batch.batch_size]
+        y = batch.y[:batch.batch_size].squeeze().to(torch.long)
+        loss = F.cross_entropy(out, y)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += float(loss)
+        total_correct += int(out.argmax(dim=-1).eq(y).sum())
+        pbar.update(batch.batch_size)
+
+    pbar.close()
+
+    loss = total_loss / len(train_loader)
+    approx_acc = total_correct / split_idx['train'].size(0)
+
+    return loss, approx_acc
+
+
+@torch.no_grad()
+def test(loader: NeighborLoader, val_steps=None):
+    model.eval()
+
+    total_correct = total_examples = 0
+    for i, batch in enumerate(loader):
+        if val_steps is not None and i >= val_steps:
+            break
+        batch = batch.to(device)
+        batch_size = batch.num_sampled_nodes[0]
+        out = model(batch.x, batch.edge_index)[:batch_size]
+        pred = out.argmax(dim=-1)
+        y = batch.y[:batch_size].view(-1).to(torch.long)
+
+        total_correct += int((pred == y).sum())
+        total_examples += y.size(0)
+
+    return total_correct / total_examples
+
+
+test_accs = []
+val_accs = []
+times = []
+train_times = []
+inference_times = []
+best_val = best_test = 0.
+prep_time = round(time.perf_counter() - wall_clock_start, 2)
+print("Total time before training begins (prep_time)=", prep_time, "seconds")
+print("Beginning training...")
+for run in range(1, num_runs + 1):
+    start = time.time()
+    print(f'\nRun {run:02d}:\n')
+
+    model.reset_parameters()
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
+
+    best_val_acc = best_test_acc = 0.0
+    for epoch in range(1, num_epochs + 1):
+        train_start = time.time()
+        loss, acc = train(epoch)
+        train_end = time.time()
+        train_times.append(train_end - train_start)
+        print(f'Epoch {epoch:02d}, Loss: {loss:.4f}, Approx. Train: {acc:.4f}'
+              f' Time: {train_end - train_start:.4f}s')
+
+        inference_start = time.time()
+        train_acc = test(test_loader)
```

Comment:
What is `test(test_loader)`?

Reply:
test function is used to calculate the accuracy on particular data, training data, validation data or testing data.